### ISSUE TITLE
Implement AMD and Intel GPU Pipeline Through torch-directml

### ISSUE THREAD

**Issue by NullSenseStudio** [carson-katri/dream-textures#504]
I've seen interest in adding an ONNX pipeline, but I really think torch-directml it already better for non-CUDA GPU's. They both would use Direct ML internally but there are some key differences.

- doesn't require special .onnx files that are less often distributed, and doesn't require extra steps in converting .ckpt files
- it's newer and only currently a development release so I'd expect to run into more bugs and incompatibility
- able to work in the already modified DiffusionPipelines with a couple of small patches
- 3-4 times faster than ONNX (on my machine at least, and I've read reports of how slow ONNX can be so I don't think it's a fluke)

How to run on diffusers:
1. pip install torch-directml
2. edit `diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py` and change `latents = self.scheduler.step(noise_pred, t, latents, **extra_step_kwargs).prev_sample` to `latents = self.scheduler.step(noise_pred.to('cpu'), t.to('cpu'), latents.to('cpu'), **extra_step_kwargs).prev_sample.to(latents.device)`. Various schedulers (if not all) do some operations with some tensors on cpu and others on the desired device. PyTorch can handle that okay for cuda and mps, but dml will raise an error. Simplest to defer to cpu rather than modifying each scheduler, no noticeable performance loss.
```python
import diffusers
import torch
import torch_directml

# not supported on dml right now
# uses a little more than half of inference time so I expect further speed improvement once its implemented
original_group_norm = torch.group_norm
def group_norm(input, num_groups, weight, bias, eps, affine):
    return original_group_norm(input.to('cpu'), num_groups, weight.to('cpu'), bias.to('cpu'), eps, affine).to(input.device)

torch.group_norm = group_norm

# not float16 compatible
pipe = diffusers.StableDiffusionPipeline.from_pretrained("runwayml/stable-diffusion-v1-5")
# just enough memory saving to work on 8GB VRAM without sequential CPU offload
pipe.enable_attention_slicing(1)

dml = torch_directml.device(torch_directml.default_device())
pipe.to(dml)

pipe("brick texture", num_inference_steps=25).images[0].show()
```
